src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -287,7 +309,7 @@ public static unsafe void Clear(Array array, int index, int length)
 
             int offset = index - lowerBound;
 
-            if (index < lowerBound || offset < 0 || length < 0 || (uint)(offset + length) > (nuint)array.LongLength)
+            if (index < lowerBound || offset < 0 || length < 0 || (uint)(offset + length) > array.NativeLength)
```

Comment:
```suggestion
            if (index < lowerBound || (offset | length) < 0 || (uint)(offset + length) > array.NativeLength)
```
Ideally the JIT could do this optimization.

Reply:
Happily, already in progress! :)
https://github.com/dotnet/runtime/issues/13573